### PR TITLE
ci: project version to 3.1.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by sdelamo**
- Micronaut build to 5.3.14
- project version to 3.1.0-SNAPSHOT
- Micronaut to 3.5.4
- Micronaut Gradle Plugin to 3.5.1
- Github core branch to 3.6.x

### CODE REVIEW

**Comment by sdelamo** [comment]
@timyates can you investigate the failure?

**Comment by timyates** [comment]
@melix is this right?  I had to change the builds from;

- `testNativeImage` to `nativeTestCompile`
- `nativeImage` to `nativeCompile`

And set the args as `buildArgs` in a block like so:

```
    graalvmNative {
        binaries {
            main {
                 ...
            }
        }
    }
```

**Review by melix — CHANGES_REQUESTED** [review]

**Comment by timyates** [comment]
Awesome, thanks @melix 

Done I think

I removed the onlyIf weirdness for H2 by only adding the minimal plugin to that project which I think does the same thing

File: test-graalvm/build.gradle [hunk 1]
```diff
@@ -28,7 +29,11 @@ subprojects { Project subproject ->
         return
     }
 
-    apply plugin: 'io.micronaut.application'
+    if (project.name == 'h2') {
```
Comment (melix) [inline test-graalvm/build.gradle:32]: Ideally there shouldn't be an `if`, but both projects should apply a different plugin.

**Review by melix — APPROVED** [review]

**Comment by timyates** [comment]
Hmmm...mariadb failures with graal...  I'm having flashbacks

**Comment by melix** [comment]
This is because the `nativeCompile` task was skipped, for some reason: https://ge.micronaut.io/s/mr3oz4o7w3vws/timeline?details=jn27hsslv3oiq

**Comment by timyates** [comment]
Great spot!  Investigating 🔍 🕵️

**Pull request merged**